pkg/cli/cmd/bicep/publish/publish.go

```diff
@@ -167,8 +171,30 @@ func (r *Runner) Run(ctx context.Context) error {
 	r.Destination = dest
 
 	err = r.publish(ctx)
-	if err != nil {
-		return clierrors.MessageWithCause(err, "Failed to publish Bicep file %q to %q.", r.File, r.Target)
+
```

Comment:
This is looking a lot better overall 👍 

This code will still need to handle the case where the error is NOT `*errcode.ErrorResponse`. Otherwise cases like a network failure will be reported as success.

You might also want to refactor the error handling for `httpErr` into it's own function for readability.

It might look something like this:

```go
err = r.publish(ctx)
var httpErr *errcode.ErrorResponse
if errors.As(err, &httpErr) {
  return handleErrorResponse(httpErr)
} else if err != nil {
  return clierrors.MessageWithCause(err, "Failed to publish Bicep file %q to %q.", r.File, r.Target)
}
```

```
func handleErrorResponse(err *errcode.ErrorResponse) error {
... // Your current code goes here.
}
```

Reply:
Changed made as requested. 

> This code will still need to handle the case where the error is NOT *errcode.ErrorResponse.
OMG 💯